api/tests/unit/utils/merge-permissions.test.ts

```diff
@@ -0,0 +1,122 @@
+import { mergePermission } from '../../../src/utils/merge-permissions';
+import { Permission, Filter } from '@directus/shared/types';
+
+const fullFilter = {} as Filter;
+const conditionalFilter = { user: { id: { _eq: '$CURRENT_USER' } } } as Filter;
+
+const permissionTemplate = {
+	role: null,
+	collection: 'directus_users',
+	permissions: null,
+	validation: null,
+	presets: null,
+	fields: null,
+} as Permission;
+
+describe('merging permissions', () => {
+	test('processes _or permissions', () => {
+		const mergedPermission = mergePermission(
+			'or',
+			{ ...permissionTemplate, permissions: conditionalFilter },
+			{ ...permissionTemplate, permissions: conditionalFilter }
+		);
+		expect(mergedPermission).toStrictEqual({
+			...permissionTemplate,
+			permissions: {
+				_or: [conditionalFilter, conditionalFilter],
+			},
+		});
+	});
```

Comment:
You're using the same input twice, which means that this test won't catch it when the `mergePermission` function accidentally returns the first permission twice. Lets tweak this to have two different test permissions to test with 🙂

Reply:
Added. Thank you!